### PR TITLE
Server

### PR DESCRIPTION
**Pull request by flxbe**
The server was restructured to extract the requests and jobs to their own files. This way, new jobs can be added without knowing about the rest of the system.

### CODE REVIEW

File: SERVER_ARCH.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Server Architecture
+
+The server is responsible for controlling all objects on the map. The player
+can only has indirect control.
```
Comment (kruesemann) [inline SERVER_ARCH.md]: -can

File: app/html-gui/window.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    this.shadow = options.shadow || false;
+
+    this.borders = options.borders || false;
+    this.node.style.borderColor = "#dad9da";
+    this.node.style.borderStyle = "solid";
+
+    this.x = 0;
+    this.y = 0;
+
+    this._onDrag = event => this.onDrag(event);
```
Comment (kruesemann) [inline app/html-gui/window.js]: Shouldn't
```js
this._onDrag = this.onDrag;
```
work?

Comment (flxbe) [reply app/html-gui/window.js]: A normal function loses the `this` context when transfered as an argument. Using this lambda function, the value for `this` is saved in the scope of the lambda function.

**Comment by flxbe** [comment]
Should all be fixed.

**Pull request merged**